aspnetcore/mobile/native-mobile-backend.md

```diff
@@ -75,29 +75,103 @@ Configure the implementation in `Startup.cs`:
+The controller requires an `ITodoRepository` to function; request an instance of this type through the controller's constructor. At runtime, this instance is provided using the framework's support for [dependency injection](../fundamentals/dependency-injection.md).
 
 :::code language="csharp" source="~/../xamarin-forms-samples/WebServices/TodoREST/TodoAPI/TodoAPI/Controllers/TodoItemsController.cs" id="snippetDI":::
 
-This API supports four different HTTP verbs to perform CRUD (Create, Read, Update, Delete) operations on the data source. The simplest of these is the Read operation, which corresponds to an HTTP GET request.
+This API supports four different HTTP verbs to perform CRUD (Create, Read, Update, Delete) operations on the data source. The simplest of these is the Read operation, which corresponds to an HTTP `GET` request.
+
+## Test the API using curl
+
+You can test the API method using a variety of tools. For this tutorial the following open source command-line tools are used:
+
+* [curl](https://curl.se/): Transfers data using various protocols including HTTP and HTTPS. curl is used in this tutorial to call the API using HTTP methods `GET`, `POST`, `PUT` and `DELETE`.
+* [jq](https://jqlang.github.io/jq/download/): A JSON processor used in this tutorial to format JSON data so that it's easy to read from the API response.
+
+### Install curl and jq on macOS
+
+On macOS, curl is preinstalled and used directly in the Terminal application. jq can be installed from Homebrew from the terminal:
+
+Install Homebrew, if not already installed, with the following command:
+
+```Bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+
+Follow the instructions presented by the installer.
+
+Install jq using Homebrew with the following command:
+
+```Bash
+brew install jq
+```
+
+For more details on Homebrew and jq installation, see [Homebrew](https://brew.sh) and [jq](https://jqlang.github.io/jq/download/).
+
+### Install curl and jq on Windows
+
+Git Bash provides a Unix-style environment on Windows, includes curl and is open-source. Follow the install steps for [Git for Windows from the official site](https://gitforwindows.org). Git for Windows includes Git Bash, which comes with curl preinstalled.
+
+Once Git Bash is installed, right click on the Git Bash application and select **Run as administrator**.
+
+In the Git Bash window, run the following command to install jq:
+
+```Bash
+curl -L -o /usr/bin/jq.exe https://github.com/stedolan/jq/releases/latest/download/jq-win64.exe
+```
+
+For more details on Git Bash and jq installation, see [Git for Windows](https://gitforwindows.org) and [jq](https://jqlang.github.io/jq/download.)
```

Comment:
Can this be made into a zone pivot?

The instructions for macOS are probably fine, but GitBash technically isn't required for Windows. Windows already has curl installed and [jq can be installed with WinGet](https://jqlang.github.io/jq/download/) (which is also [already in Windows](https://learn.microsoft.com/en-us/windows/package-manager/winget/)).

So couldn't you just have them launch PowerShell on their machine, install jq with `winget install jqlang.jq`, and then go to town?

Reply:
@CamSoper, you are right!  Originally, I was trying to find a way to keep the commands for curl the same for macOS/Unix-style bash vs running curl on windows.  For these old topic and old versioned sample I wanted to be as non-intrusive to the doc and samples as I could.  But going with jq for formatting changed that so I could do the same for both without suggesting a Git Bash which came with curl.

I will change it as you suggest to use curl that comes with windows 10 + and use jq for that to keep the commands the same. (instead of `| Select-String -Pattern '{' | ConvertFrom-Json`).

By "pivot" I will assume you meant tabs for each environment, but if you meant something else let me know.

Reply:
...also, going with the command line for Windows instead of PowerShell for this example.  Even though curl is installed with later versions of Windows, PowerShell does not directly use it.  Instead, it interprets `curl` as an alias for the `Invoke-WebRequest` cmdlet, not the actual curl.exe and you start to run into to some minor differences.  So in the spirit of keeping this as low in overheads steps as possible will instruct to use the command line even though it is not friendly to multiline formatting for commands.